doc/cookbook/named-routes/NamedRoutes.lhs

```diff
@@ -0,0 +1,393 @@
+                                 :<|> ((MovieId -> Movie -> Handler NoContent)
+                                       :<|> (MovieId -> Handler NoContent))))
+    • In the expression:
+        versionHandler
+          :<|>
+            movieListHandler
+              :<|>
+                getMovieHandler :<|> updateMovieHandler :<|> deleteMovieHandler
+      In an equation for 'server':
+          server
+            = versionHandler
+                :<|>
+                  movieListHandler
+                    :<|>
+                      getMovieHandler :<|> updateMovieHandler :<|> deleteMovieHandler
+    |
+226 | server = versionHandler
+    |
+```
+On the contrary, with the record-based technique, we refer to the routes by their name:
+```haskell,ignore
+data API mode = API
+    { list   :: "list" :> ...
+    , delete ::  "delete" :> ...
+    }
+```
+and GHC follows the lead:
+```console
+    • Couldn't match type 'NoContent' with 'Movie'
+      Expected type: AsServerT Handler :- Delete '[JSON] Movie
+        Actual type: Handler NoContent
+    • In the 'delete' field of a record
+      In the expression:
+        MovieAPI
+          {get = getMovieHandler movieId,
+           update = updateMovieHandler movieId,
+           delete = deleteMovieHandler movieId}
+      In an equation for 'movieHandler':
+          movieHandler movieId
+            = MovieAPI
+                {get = getMovieHandler movieId,
+                 update = updateMovieHandler movieId,
+                 delete = deleteMovieHandler movieId}
+    |
+252 |     , delete = deleteMovieHandler movieId
+    |
+```
+
+So, records are more readable for a human, and GHC gives you more accurate error messages, so 
+why ever look back? Let's get started! 
+
+
+<details> 
+  <summary>The boilerplate required for both the nested and flat case</summary>
+
+```haskell
+{-# LANGUAGE GHC2021             #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE DerivingVia         #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+
+import Control.Exception          (throwIO)
+import Control.Monad.Trans.Reader (ReaderT, runReaderT)
+import Network.Wai.Handler.Warp   (run)
+import System.Environment         (getArgs)
+import Data.Aeson                 (FromJSON (..), ToJSON (..))
+import GHC.Generics               (Generic, Generically (..))
+import Data.List                  (sortOn)
+import Data.Text                  (Text)
+import Data.Foldable              (find)
+
+import Servant 
+import Servant.Client 
+import Servant.Client.Generic 
+import Servant.Server
+import Servant.Server.Generic
+```
+
+</details>
+
+## Domain context
+
+Consider a `Movie` constructed from a `Title` and a `Year` of publication.
+
+```haskell
+data Movie = Movie
+    { movieId :: MovieId
+    , title :: Title
+    , year :: Year
+    }
+    deriving stock Generic
```

Comment:
Hmm, how come you didn't have to enable [DerivingStrategies](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/deriving_strategies.html#extension-DerivingStrategies)? It's not part of GHC2021.